internal/provisioner/plain/controllers/bundleinstance_controller.go

```diff
@@ -343,20 +343,23 @@ func reconcileDesiredBundle(ctx context.Context, c client.Client, bi *rukpakv1al
 	}
 
 	// check whether there's an existing Bundle that matches the desired Bundle template
-	// specified in the BI resource, and if not, generate a new matches the template.
+	// specified in the BI resource, and if not, generate a new Bundle that matches the template.
 	b := util.CheckExistingBundlesMatchesTemplate(existingBundles, bi.Spec.Template)
 	if b == nil {
 		controllerRef := metav1.NewControllerRef(bi, bi.GroupVersionKind())
+		hash := util.GenerateTemplateHash(bi.Spec.Template)
+
 		labels := bi.Spec.Template.Labels
 		if len(labels) == 0 {
 			labels = make(map[string]string)
 		}
 		labels[util.CoreOwnerKindKey] = rukpakv1alpha1.BundleInstanceKind
 		labels[util.CoreOwnerNameKey] = bi.GetName()
+		labels[util.CoreBundleTemplateHashKey] = hash
 
 		b = &rukpakv1alpha1.Bundle{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:            util.GenerateBundleName(bi.GetName()),
+				Name:            fmt.Sprintf("%s-%s", bi.GetName(), hash),
```

Comment:
I _think_ we can still have a helper function that's responsible for generating the Bundle's metadata.Name but I don't have strong opinions here.